zerver/tests/test_auth_backends.py

```diff
@@ -342,6 +342,23 @@ def test_email_auth_backend_disabled_password_auth(self) -> None:
                 )
             )
 
+    def test_email_auth_backend_weak_password_no_session(self) -> None:
+        user_profile = self.example_user("hamlet")
+        password = "a_password_of_22_chars"
+        user_profile.set_password(password)
+        user_profile.save()
+
+        with self.settings(PASSWORD_MIN_LENGTH=30):
+            try:
+                EmailAuthBackend().authenticate(
+                    request=HttpRequest(),
+                    username=self.example_email("hamlet"),
+                    password=password,
+                    realm=get_realm("zulip"),
+                )
+            except JsonableError as e:
+                self.assertEqual(str(e), "You need to reset your password.")
```

Comment:
Let's replace this with an end-to-end test of actually attempting mobile authentication, not just calling the auth backend.

Reply:
Added two tests similar to the existing `AuthBackendTest.test_email_auth_backend_password_hasher_change` and `FetchAPIKeyTest.test_old_weak_password_after_hasher_change`.

Not quite sure how to write a test for email auth with mobile. Do we have anything similar doing email auth for mobile for me to use as reference or any other pointers that'd help? Wasn't able to find any tests for mobile flow other than the ones in social auth tests.